src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -152,100 +200,146 @@ class boss_eadric : public CreatureScript
+            }
+        }
+
+        void EnterCombat(Unit* who) override
+        {
+            if (instance->GetBossState(DATA_EADRIC_THE_PURE) != SPECIAL)
+            {
+                events.ScheduleEvent(EVENT_VENGEANCE, 500);
+                events.ScheduleEvent(EVENT_RADIANCE, urand(7000, 15000));
+                Talk(SAY_AGGRO_E, who);
+                _EnterCombat();
+            }
+        }
+
+        void KilledUnit(Unit* who) override
+        {
+            Talk(SAY_KILL_UNIT_E, who);
+        }
 
-            me->DisappearAndDie();
+        void AttackStart(Unit* who) override
+        {
+            if (instance->GetBossState(DATA_EADRIC_THE_PURE) != SPECIAL)
+                BossAI::AttackStart(who);
         }
 
         void UpdateAI(uint32 uiDiff) override
         {
-            if (bDone && uiResetTimer <= uiDiff)
-            {
-                me->GetMotionMaster()->MovePoint(0, 746.87f, 665.87f, 411.75f);
-                bDone = false;
-            } else uiResetTimer -= uiDiff;
+            events.Update(uiDiff);
 
-            if (!UpdateVictim())
+            if (me->HasUnitState(UNIT_STATE_CASTING))
                 return;
 
-            if (uiHammerJusticeTimer <= uiDiff)
+            while (uint32 eventId = events.ExecuteEvent())
             {
-                me->InterruptNonMeleeSpells(true);
-
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 250, true))
+                switch (eventId)
                 {
-                    if (target && target->IsAlive())
-                    {
-                        DoCast(target, SPELL_HAMMER_JUSTICE);
-                        DoCast(target, SPELL_HAMMER_RIGHTEOUS);
-                    }
+                    case EVENT_VENGEANCE:
+                        if (!me->HasAura(SPELL_VENGEANCE))
+                            DoCast(me, SPELL_VENGEANCE);
+                        events.ScheduleEvent(EVENT_VENGEANCE, 30000);
+                        break;
+                    case EVENT_RADIANCE:
+                        DoCastAOE(SPELL_RADIANCE);
+                        Talk(EMOTE_RADIANCE);
+                        events.ScheduleEvent(EVENT_HAMMER_OF_JUSTICE, urand(20000, 30000));
+                        break;
+                    case EVENT_HAMMER_OF_JUSTICE:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100.0f, true))
+                        {
+                            DoCast(target, SPELL_HAMMER_JUSTICE, true);
+                            DoCast(target, SPELL_HAMMER_RIGHTEOUS);
+                            Talk(EMOTE_HAMMER_RIGHTEOUS, target);
+                            Talk(SAY_HAMMER_RIGHTEOUS);
+                        }
+                        events.ScheduleEvent(EVENT_RADIANCE, urand(7000, 15000));
+                        break;
+                    case EVENT_EADRIC_DONE:
+                        me->RemoveAllAuras();
+                        me->DeleteThreatList();
```

Comment:
@Appled @joschiwald this line is the cause of secound boss dont work.
When event EVENT_EADRIC_DONE is called, DeleteThreatList make creature enter in evade, and all script stuck in secound boss.
Just removind this line, instance works well (dont tested achievs)
And need update ```UNIT_FLAG_REMOVE_CLIENT_CONTROL``` flags